### PR TITLE
Added negative seconds check and error

### LINKED ISSUE #147: sleep doesn't raise error when given negative
If you give the robot.sleep function a negative time interval it doesn't raise an error like the normal time module would (`ValueError: sleep length must be non-negative`) but instead it will just wait for ever. Given how the robot.sleep and robot.time modules are designed to be as close as possible to the time module to use, it should probably raise the same error as the time module does here

### PR DESCRIPTION
**Pull request by 13ros27**
Add a check for negative seconds in the sleep function which then raises a value error which should fix #147

### CODE REVIEW

**Comment by antoinepetty** [comment]
@PeterJCLaw any idea why the CI isn't coming up here? Is it because the branch is outside the repo?

**Comment by trickeydan** [comment]
> Is it because the branch is outside the repo?

Likely. There's a box you have to tick in CircleCI to build for external repos. It's safe to turn on unless we have any secrets in the CI (e.g for CD), which I don't think we do.

I'd suggest allowing builds of PRs, so that external contributors can have tests run :)

**Comment by RealOrangeOne** [comment]
> Is it because the branch is outside the repo?

It is indeed! Have flicked the switch so it shouldn't happen in future. It won't affect this PR though until new commits are pushed.

File: modules/sr/robot/robot.py [hunk 1]
```diff
@@ -147,6 +147,10 @@ def sleep(self, secs: float) -> None:
         """
         Roughly equivalent to `time.sleep` but accounting for simulation time.
         """
+
```
Comment (trickeydan) [inline modules/sr/robot/robot.py:150]:
```suggestion

```

**Pull request merged**